src/DeadCodeElimPass.cpp

```diff
@@ -63,15 +64,16 @@ bool DeadCodeElimPass::runOnFunction(llvm::Function &F) {
   size_t deleted = 0;
   UseSet use = computeUseSets(F);
 
-  auto &BBL = F.getBasicBlockList();
-  for (auto BBI = BBL.rbegin(); BBI != BBL.rend(); ++BBI) {
+  //TODO: Find a better solution to replace the reverse iterator
+  for (auto BBI = --F.end(); ; --BBI) {
     for (auto it = BBI->end(); it != BBI->begin();) {
       llvm::Instruction &I = *--it;
       if (!(unsafeToDelete(I) || use.count(&I))) {
         it = I.eraseFromParent();
         ++deleted;
       }
     }
+    if(BBI == F.begin()) break;
```

Comment:
... and if not, you might want to use this reverse-loop idiom;
```c++
for (auto BBI = F.end(); BBI != F.begin(); --BBI) {
  --BBI;
```
, or at the very least use `std::prev(F.end())` instead of `--F.end()`.